hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java

```diff
@@ -136,6 +137,37 @@ public GasCharges chargeForGas(
         }
     }
 
+    /**
+     * Tries to charge intrinsic gas for the given transaction based on the pre-fetched sender accountID,
+     * within the given context and world updater.  This is used when transaction are aborted due to an exception check
+     * failure before the transaction has started execution in the EVM.
+     *
+     * @param sender  the sender accountID
+     * @param context the context of the transaction, including the network gas price
+     * @param worldUpdater the world updater for the transaction
+     * @param transaction the transaction to charge gas for
+     * @throws HandleException if the gas charging fails for any reason
+     */
+    public void chargeGasForAbortedTransaction(
+            @NonNull final AccountID sender,
+            @NonNull final HederaEvmContext context,
+            @NonNull final HederaWorldUpdater worldUpdater,
+            @NonNull final HederaEvmTransaction transaction) {
+        requireNonNull(sender);
+        requireNonNull(context);
+        requireNonNull(worldUpdater);
+        requireNonNull(transaction);
+
+        final var senderAccount = worldUpdater.getHederaAccount(sender);
+        requireNonNull(senderAccount);
+        final var intrinsicGas = gasCalculator.transactionIntrinsicGasCost(transaction.evmPayload(), false);
+        validateTrue(
```

Comment:
In this case should we throw or should we charge the balance left even if less than intrinsic. 
I guess right now that's minimal hbar but still

Reply:
We could do that.  There should have been checks to prevent this from occurring earlier but it is possible that the balance has changed in the mean time.